wpengine/faustjs
# Handle `/robots.txt` requests in `handleSitemapRequests`

## Description

This PR addresses handling of the `/robots.txt` route commonly needed with custom sitemap paths. This PR adds the `robotsTxt` optional property to the `handleSitemapRequests` config object.

This property is a callback function that, when defined, handles the appropriate `/robots.txt` route, and renders the text returned from this callback function:

```tsx
handleSitemapRequests(req, {
    wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
    sitemapIndexPath: '/wp-sitemap.xml',
    async robotsTxt(sitemapUrl) {
      return `
        User-agent: *
        Allow: /

        Sitemap: ${sitemapUrl}
      `;
    },
});
```

This callback also has a `sitemapUrl` param, which can be used when defining your robots.txt content.

## Testing

Tests have been added.

### Manual Testing

1. `npm run build` at the monorepo root
2. Add the following middleware to `src/pages/_middleware.ts`:
```tsx
import { handleSitemapRequests } from '@faustjs/next/middleware';
import { NextRequest, NextResponse } from 'next/server';

export default async function middleware(req: NextRequest) {
  const isSitemapRequest = await handleSitemapRequests(req, {
    wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
    sitemapIndexPath: '/wp-sitemap.xml',
    async robotsTxt(sitemapUrl) {
      return `
        User-agent: *
        Allow: /

        Sitemap: ${sitemapUrl}
      `;
    },
  });

  if (isSitemapRequest) {
    return isSitemapRequest;
  }

  return NextResponse.next();
}
```
3. Observe the `/robots.txt` route



## Screenshots

Before adding the `robotsTxt` config:

<img width="1792" alt="Screen Shot 2022-03-04 at 8 20 50 AM" src="https://user-images.githubusercontent.com/5946219/156780516-592b72fc-68c5-4d94-95c6-ea91cd51e443.png">

After adding the `robotsTxt` config:

<img width="1792" alt="Screen Shot 2022-03-04 at 8 20 35 AM" src="https://user-images.githubusercontent.com/5946219/156780562-5cd011a3-a7d8-48b9-81b0-417333a7b927.png">




## Documentation Changes



## Dependant PRs

- #790
